mne/utils/_bunch.py

```diff
@@ -94,10 +96,58 @@ def _named_subclass(klass):
 class NamedInt(_Named, int):
     """Int with a name in __repr__."""
 
-    pass
-
 
 class NamedFloat(_Named, float):
     """Float with a name in __repr__."""
 
-    pass
+
+class MNEObjectsList(MutableSequence):
```

Comment:
One problem I expect with this subclassing `MutableSequence` instead of `list` is that `mne.externals.h5io.write_hdf5(...)` will not work on it. IIRC currently you can `write_hdf5` on `info` objects and it will work because all `info` entries and Python built-in types. After this PR it will not. Can you check? If this is the case we should fix it somehow.

I would consider just subclassing `list` like we do for SourceSpaces, and if at some point we want to not subclass list for these sorts of things, we can change `Digitization` and `SourceSpaces` at the same time.

Reply:
I have the same feeling as @larsoner here. Subclassing list is not a great pattern but it's simple. We just want to have a repr. The argument to make sure that it contains only instance of DigPoint is fair but do we really expect to have the problem with users hacking the info['dig'] manually? I don't expect anyone to do this and think that it will be fine...

my 2c

Reply:
I can make `MNEObjectsList` derive from `list()` I can leave with it.

I am not expecting "users" to mess up the list, but I do like to have some safety nets when writing digitizations, montages, etc.. I can always get read of this class and do that in Digitization, or keep it as a private thing. But we have lots of lists of things and we never ensure those elements are always the same. (if I'm not mistaken, this object does not prevent insertions of other types. I thought YAGNI, and that it was easy to add if needed)

Reply:
can we avoid the MNEObjectsList and have Digitization inherit from list? it would remove a lot of code in this PR

Reply:
It's a trade off between safety net and barrier to write new code. The more you nest classes, the less accessible it becomes to new contributors. I tend to lean more on the side of assuming good intent -- i.e., if you hack info['dig'] you are on your own.